From ce4fbc288cb2d5d017758cba2793adeb3077c3a1 Mon Sep 17 00:00:00 2001
From: Junhao He <hejunhao3@huawei.com>
Date: Mon, 12 Dec 2022 20:27:57 +0800
Subject: [PATCH] coresight: trbe: Enable ACPI/Platform automatic module
 loading
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: ce4fbc288cb2d5d017758cba2793adeb3077c3a1
Modified-by-SEL: No


driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5YCYK

--------------------------------------------------------------------------

Lets add the MODULE_TABLE and platform id_table entries so that
the TRBE driver can attach to the ACPI platform device created by
the acpi trbe code.

Signed-off-by: Junhao He <hejunhao3@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/hwtracing/coresight/coresight-trbe.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/hwtracing/coresight/coresight-trbe.c b/drivers/hwtracing/coresight/coresight-trbe.c
index a84f4a51e08a..881a89e245fb 100644
--- a/drivers/hwtracing/coresight/coresight-trbe.c
+++ b/drivers/hwtracing/coresight/coresight-trbe.c
@@ -1189,7 +1189,14 @@ static const struct of_device_id arm_trbe_of_match[] = {
 };
 MODULE_DEVICE_TABLE(of, arm_trbe_of_match);
 
+static const struct platform_device_id arm_trbe_match[] = {
+	{ ARMV9_TRBE_PDEV_NAME, 0},
+	{}
+};
+MODULE_DEVICE_TABLE(platform, arm_trbe_match);
+
 static struct platform_driver arm_trbe_driver = {
+	.id_table = arm_trbe_match,
 	.driver	= {
 		.name = DRVNAME,
 		.of_match_table = of_match_ptr(arm_trbe_of_match),
-- 
2.33.0

